Mathlib/NumberTheory/FactorisationProperties.lean

```diff
@@ -0,0 +1,248 @@
+  simp only [Nat.WeirdNumber, not_and_or]
+  left
+  exact prime_not_abundant n h
+
+theorem not_pseudoperfect_not_perfect (h : ¬ Pseudoperfect n) : ¬ Perfect n := by
+  revert h
+  rw [not_imp_not]
+  exact perfect_pseudoperfect n
+
+theorem prime_not_pseudoperfect (h : Nat.Prime n) : ¬ Nat.Pseudoperfect n := by
+  rw [not_pseudoperfect_iff_forall]
+  have h1 : powerset (properDivisors n) = {∅, {1}} := by
+    rw [Prime.properDivisors h]
+    exact rfl
+  rw [h1]
+  right
+  rintro s hs
+  rcases (List.mem_pair.mp hs) with hs | hs <;>
+  {simp only [ne_eq, hs, sum_empty, sum_singleton];
+    exact Nat.ne_of_lt (by linarith [Prime.one_lt h])}
+
+theorem prime_not_perfect (h : Prime n) : ¬ Perfect n :=
+  not_pseudoperfect_not_perfect n (prime_not_pseudoperfect n h)
+
+theorem prime_deficient (h : Prime n) : Deficient n := by
+  rw [deficient_not_abundant_or_perfect n]
+  · exact ⟨prime_not_abundant n h, prime_not_perfect n h⟩
+  · exact Prime.pos h
+
+/-There exists infinite deficient numbers-/
+theorem exists_infinite_deficients : ∃ d, n ≤ d ∧ Deficient d := by
+  obtain ⟨p, ⟨h1, h2⟩⟩ := exists_infinite_primes n
+  have Deficientp : Deficient p :=  prime_deficient p h2
+  use p
+
+/-Any natural number power of a prime is deficient-/
+theorem prime_pow_deficient (h : Prime n) : Deficient (n^m) := by
+  obtain hL | hR := Nat.eq_zero_or_pos m
+  · rw [hL, Nat.pow_zero]
+    exact le.refl
+  · have n_gtOne : 1 < n := Prime.two_le h
+    have hp : properDivisors (n ^ m) = image (n ^ ·) (range m) := by
+      apply subset_antisymm <;> intro a
+      · simp only [mem_properDivisors, mem_image, mem_range, dvd_prime_pow h]
+        rintro ⟨⟨t, ht, rfl⟩, ha'⟩
+        exact ⟨t, lt_of_le_of_ne ht (fun ht' ↦ lt_irrefl _ (ht' ▸ ha')), rfl⟩
+      · simp only [mem_image, mem_range, mem_properDivisors, forall_exists_index, and_imp]
+        intro x hx hy
+        constructor
+        · rw [← hy, dvd_prime_pow h]
+          use x
+          exact ⟨Nat.le_of_succ_le hx, rfl⟩
+        · rw [← hy]
+          exact (Nat.pow_lt_pow_iff_right (Prime.two_le h)).mpr hx
+    have hw : ∑ i in image (fun x => n ^ x) (range m), i = ∑ i in range m, n^i := by
+      rw [Finset.sum_image]
+      rintro x _ y _ hnxy
+      by_contra hc
+      rcases (Ne.lt_or_lt hc) with hxy1 | hxy2
+      · linarith [(pow_lt_pow_iff_right n_gtOne).mpr hxy1]
+      · linarith [(pow_lt_pow_iff_right n_gtOne).mpr hxy2]
+    have hq : ∑ i in range m, n ^ i = (n ^ m - 1) / (n - 1) := Nat.geomSum_eq (Prime.two_le h) _
+    have hr : 1 < n ^ m :=  one_lt_pow (not_eq_zero_of_lt hR) (Prime.two_le h)
+    rw [Deficient, hp, hw, hq]
+    calc
+      (n ^ m - 1) / (n - 1) ≤ (n ^ m - 1) := Nat.div_le_self (n ^ m - 1) (n - 1)
+      _                   < n ^ m := by {rw [tsub_lt_self_iff]; norm_num; exact lt_of_succ_lt hr}
+
+theorem exists_infinite_even_deficients : ∃ d, n ≤ d ∧ Deficient d ∧ Even d := by
+  use 2 ^ (n + 1)
+  constructor
+  · rw [Nat.le_iff_lt_or_eq]
```

Comment:
Does mathlib have this estimate already? If not, please extract this into a separate lemma, for any base `a : N`.

Reply:
There's `Nat.lt_two_pow` which is considerably stronger, it shows `n < 2 ^ n`. This is actually used in the proof here, although in a strangely roundabout way.